observability-test/spanner.ts

```diff
@@ -1298,3 +1320,967 @@ describe('E2E traces with async/await', async () => {
+      runSpan.spanContext().traceId,
+      runStreamSpan.spanContext().traceId,
+      'Expected that both spans share a traceId'
+    );
+    assert.ok(
+      runStreamSpan.spanContext().spanId,
+      'Expected that runStreamSpan has a defined spanId'
+    );
+    assert.ok(
+      runSpan.spanContext().spanId,
+      'Expected that runSpan has a defined spanId'
+    );
+
+    const databaseBatchCreateSessionsSpan = spans[0];
+    assert.strictEqual(
+      databaseBatchCreateSessionsSpan.name,
+      'CloudSpanner.Database.batchCreateSessions'
+    );
+    const sessionPoolCreateSessionsSpan = spans[1];
+    assert.strictEqual(
+      sessionPoolCreateSessionsSpan.name,
+      'CloudSpanner.SessionPool.createSessions'
+    );
+    assert.ok(
+      sessionPoolCreateSessionsSpan.spanContext().traceId,
+      'Expecting a defined sessionPoolCreateSessions traceId'
+    );
+    assert.deepStrictEqual(
+      sessionPoolCreateSessionsSpan.spanContext().traceId,
+      databaseBatchCreateSessionsSpan.spanContext().traceId,
+      'Expected the same traceId'
+    );
+    assert.deepStrictEqual(
+      databaseBatchCreateSessionsSpan.parentSpanId,
+      sessionPoolCreateSessionsSpan.spanContext().spanId,
+      'Expected that sessionPool.createSessions is the parent to db.batchCreassionSessions'
+    );
+
+    // Assert that despite all being exported, SessionPool.createSessions
+    // is not in the same trace as runStream, createSessions is invoked at
+    // Spanner Client instantiation, thus before database.run is invoked.
+    assert.notEqual(
+      sessionPoolCreateSessionsSpan.spanContext().traceId,
+      runSpan.spanContext().traceId,
+      'Did not expect the same traceId'
+    );
+
+    // Ensure that the last span has an error.
+    assert.deepStrictEqual(
+      runStreamSpan.status.code,
+      SpanStatusCode.ERROR,
+      'Expected an error status'
+    );
+
+    const want = '3 INVALID_ARGUMENT: ' + messageBadSelect1p;
+    assert.deepStrictEqual(
+      runStreamSpan.status.message,
+      want,
+      `Mismatched status message:\n\n\tGot:  '${runStreamSpan.status.message}'\n\tWant: '${want}'`
+    );
+
+    // Finally check for the collective expected event names.
+    const expectedEventNames = [
+      'Requesting 25 sessions',
+      'Creating 25 sessions',
+      'Requested for 25 sessions returned 25',
+      'Starting stream',
+      'Acquiring session',
+      'Waiting for a session to become available',
+      'Acquired session',
+      'Using Session',
+    ];
+    assert.deepStrictEqual(
+      actualEventNames,
+      expectedEventNames,
+      `Unexpected events:\n\tGot:  ${actualEventNames}\n\tWant: ${expectedEventNames}`
+    );
+  }
+
+  it('database.run with bad syntax: async/await', async () => {
+    const instance = spanner.instance('instance');
+    const database = instance.database('database');
+
+    try {
+      const [rows] = await database.run(selectSql1p);
+    } catch (e) {
```

Comment:
can't we simply remove the catch block from here?

Reply:
The comment in that clause explains why we have the catch, to keep the code in semblance of being copied directly from customer code but even more so that we can properly assert on produced spans.
```javascript
      // This catch is meant to ensure that we
      // can assert on the generated spans.
```